velox/functions/prestosql/tests/SequenceTest.cpp

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox;
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class SequenceTest : public FunctionBaseTest {
+ protected:
+  void testExpression(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const VectorPtr& expected) {
+    auto result = evaluate(expression, makeRowVector(input));
+    assertEqualVectors(expected, result);
+  }
+
+  void testExpressionWithError(
+      const std::string& expression,
+      const std::vector<VectorPtr>& input,
+      const std::string& expectedError) {
+    VELOX_ASSERT_THROW(
+        evaluate(expression, makeRowVector(input)), expectedError);
+  }
+};
+} // namespace
+
+TEST_F(SequenceTest, sequence) {
+  const auto startVector = makeNullableFlatVector<int>({1, 2, 10});
+  const auto endVector = makeNullableFlatVector<int>({2, 5, 9});
+  VectorPtr expected = makeNullableArrayVector<long long>({
+      {{1, 2}},
+      {{2, 3, 4, 5}},
+      std::nullopt,
+  });
+  testExpression("sequence(C0, C1)", {startVector, endVector}, expected);
+}
+
+TEST_F(SequenceTest, testConstant) {
+  const auto endVector = makeNullableFlatVector<int>({2, 5});
+  VectorPtr expected = makeArrayVector<long long>({{1, 2}, {1, 2, 3, 4, 5}});
+  testExpression("sequence(1, C0)", {endVector}, expected);
+}
+
+TEST_F(SequenceTest, testInvalidResult) {
+  const auto startVector = makeNullableFlatVector<int>({1});
+  const auto endVector = makeNullableFlatVector<int>({100000});
+  testExpressionWithError(
+      "sequence(C0, C1)",
+      {startVector, endVector},
+      "etConnectorFactory");
```

Comment:
This error message is rather strange. It would be nice to ensure that error message is user-friendly.